src/frontend/src/lib/schema/transaction.schema.ts

```diff
@@ -0,0 +1,17 @@
+import { z } from 'zod';
+
+const commonTypes = ['send', 'receive'] as const;
+
+const ethSpecificTypes = ['withdraw', 'deposit'] as const;
+
+const icpSpecificTypes = ['approve', 'burn', 'mint'] as const;
+
+const allTypes = [...commonTypes, ...ethSpecificTypes, ...icpSpecificTypes] as const;
+
+export const btcTransactionTypes = commonTypes;
+
+export const ethTransactionTypes = [...commonTypes, ...ethSpecificTypes];
+
+export const icpTransactionTypes = [...commonTypes, ...icpSpecificTypes];
```

Comment:
wondering if all the above shall go in folder `constants`...

Reply:
what about `z.enum` instead of const?

Reply:
done!

Reply:
git push

Reply:
(looks like you did not)

Reply:
ah sorry, check now please

Reply:
my bad it was already good, sorry. I think it's good to keep the constant there, they are only meant for zod.